drivers/power/lt8491/iio_lt8491.c

```diff
@@ -0,0 +1,361 @@
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include <errno.h>
+#include "iio_lt8491.h"
+#include "lt8491.h"
+#include "no_os_alloc.h"
+#include "no_os_error.h"
+
+#define LT8491_IIO_CH_ATTR_RW(_name, _priv)	\
+	{					\
+		.name = _name,			\
+		.priv = _priv,			\
+		.show = lt8491_ch_attr_show,	\
+		.store = lt8491_ch_attr_store,	\
+	}
+
+#define LT8491_IIO_CH_ATTR_RO(_name, _priv)	\
+	{					\
+		.name = _name,			\
+		.priv = _priv,			\
+		.show = lt8491_ch_attr_show,	\
+		.store = NULL,			\
+	}
+
+#define LT8491_IIO_CH_ATTR_WO(_name, _priv)	\
+	{					\
+		.name = _name,			\
+		.priv = _priv,			\
+		.show = NULL,			\
+		.store = lt8491_ch_attr_store,	\
+	}
+
+enum lt8491_attr_priv {
+	LT8491_TBAT,
+	LT8491_POUT,
+	LT8491_PIN,
+	LT8491_EFF,
+	LT8491_IOUT,
+	LT8491_IIN,
+	LT8491_VBAT,
+	LT8491_VIN,
+	LT8491_CHARGING_STAGE,
+	LT8491_CHARGING_STATUS,
+	LT8491_ENABLE,
+	LT8491_RESTART,
+	LT8491_SCRATCH,
+	LT8491_SERIAL_ID,
+};
+
+static int32_t lt8491_iio_reg_read(struct lt8491_iio_device *dev,
+				   uint32_t reg, uint32_t *readval)
+{
+	return lt8491_reg_read(dev->dev, (uint8_t)reg, (uint16_t*)readval,
```

Comment:
You will have to use a `uint16_t *` variable in order to read the value. Otherwise the result will be different depending on CPU endianess.